### DISCUSSION TITLE
能不能出个官方的一键安装脚本啊

### DISCUSSION THREAD

**Discussion by itdsgithub** [apernet/hysteria#708] [General]
现在用的hysteria2别人的一键按照脚本协议，总是各种各样的问题，脚本上配置似乎也不全。
能不能出个官方的脚本啊，尤其是服务器配置，生成链接之类的，适合我们这些小白用户。

**Comment by tobyxdd** [comment]
https://v2.hysteria.network/zh/docs/getting-started/Installation/#linux

**Reply by itdsgithub** [reply]
小白表示就是这些服务器相关参数不会配置，想要类似这样的脚本https://blog.misaka.rest/2023/09/02/hysteria2-script/
但是他的脚本，参数不全，启动后各种问题

**Reply by haruue** [reply]
安装/升级 Hysteria， 与生成 Hysteria 配置， 这是两件事情。

这个需求应该是要做成一个 **交互式的**， 用来生成服务器配置的静态页面。 这个页面提供一个表单， 上面有所有可选项以及对这些选项的解释。 用户在填写这个表单后， 就能生成配置文件（如同用户在使用 OpenWRT LuCI）， 并允许用户复制或者下载然后用 WinSCP 这样的图形化工具来部署到服务器上。

**Reply by CussiusKon** [reply]
类似x-ui那种的吧，确实比较方便

**Reply by itdsgithub** [reply]
对对，但是现在也没见到谁做出来基于hysteria2的交互式页面，类似x-ui那种

**Reply by MiniKoro** [reply]
就应该像sing-box作者一样 无视小白

**Reply by itdsgithub** [reply]
干嘛设置那么高门槛，不要歧视我们小白，我们会翻个墙都已经很厉害了

**Reply by MiniKoro** [reply]
不是歧视小白，主要人家文档已经写的非常清晰门槛已经足够低了，开发者能持续输出核心的东西就够了，还要人家搞个可视化操作界面，真的很离谱啊

**Reply by 1126188417** [reply]
这还是看开发者的兴趣吧，开发者可能也不知道咱们到底有多白，那些 youtuber 靠流量吃饭他们肯定更乐意做一键脚本，他们自己会慢慢更新的。😁

**Reply by TuSinaWorld** [reply]
> 干嘛设置那么高门槛，不要歧视我们小白，我们会翻个墙都已经很厉害了

并不是歧视小白,而是圈子一但大了非常容易出现问题,clash火上微博算是前车之鉴.而且这种东西是开发者无私奉献做出的东西,没有任何人有资格能指责他们.

> 对对，但是现在也没见到谁做出来基于hysteria2的交互式页面，类似x-ui那种

至于一键脚本,说实话,hy2是我知道最容易安装的协议了,运行官方bash脚本后将配置文件中域名改成你自己的(确保域名解析正确),最后systemctl enable hysteria-server.service和systemctl start hysteria-server.service就行了,全过程不到五分钟,x-ui安装个vmess都不一定比这快.我个人认为这种极简化的协议实在没必要一键脚本.

最后,建议多看文档,有些东西总得你努力才能学会,不是吗?

**Reply by gsuns** [reply]
我认为你不愿意帮可以不帮 这是你的自由.
就拿我父母来说,用手机都困难,叫他用电脑打字做文档,堪比登天.
对于无能为力 不会配置 hysteria 的人,他们需要帮助我认为这件事本身没有错 .因此不应该宣扬 [无视需要帮助的人] 的这种文化.

**Comment by gsuns** [comment]
你说的脚本我不太懂,我一般都是把配置保存在 在线笔记本上,下次如果要用直接粘贴过去.
我不知道你哪里出了问题,如果是按照官方文档来配置.如果是粘贴配置这种方法,几分钟就完成了

1.在linux终端输入来安装
```
bash <(curl -fsSL https://get.hy2.sh/)
```
2.编辑HY2配置文件,路径是
```
/etc/hysteria/config.yaml
``` 

然后根据你的需要,查看https://v2.hysteria.network/zh/docs/advanced/Full-Server-Config/ 去配置

3. 启动
```
systemctl start hysteria-server.service
```
4.加入开机启动项
```
systemctl enable hysteria-server.service
```

查看HY2状态
```
systemctl status hysteria-server.service
```
